ProcessLib/Utils/ProcessUtils.cpp

```diff
@@ -43,17 +44,66 @@ std::vector<std::reference_wrapper<ProcessVariable>> findProcessVariables(
     std::vector<ProcessVariable> const& variables,
     BaseLib::ConfigTree const& pv_config,
     std::initializer_list<std::string>
-        tag_names)
+        tags)
 {
     std::vector<std::reference_wrapper<ProcessVariable>> vars;
-    vars.reserve(tag_names.size());
+    vars.reserve(variables.size());
 
-    for (auto const& tag : tag_names)
+    if (variables.size() > tags.size())
+        DBUG("Found multiple process variables with a same tag.");
+
+    for (auto& tag : tags)
     {
-        vars.emplace_back(findProcessVariable(variables, pv_config, tag));
+        auto vars_per_tag = findProcessVariables(variables, pv_config, tag);
+        vars.insert(vars.end(), vars_per_tag.begin(), vars_per_tag.end());
     }
 
     return vars;
 }
 
+std::vector<std::reference_wrapper<ProcessVariable>> findProcessVariables(
+    std::vector<ProcessVariable> const& variables,
+    BaseLib::ConfigTree const& pv_config,
+    std::string const& tag)
+{
+    std::vector<std::reference_wrapper<ProcessVariable>> vars;
+
+    auto var_names = pv_config.getConfigParameterList<std::string>(tag);
+
+    if (var_names.empty())
+        OGS_FATAL("Config tag <%s> is not found.", tag.c_str());
+
+    std::vector<std::string> cached_var_names;
+
+    for (std::string const& var_name : var_names)
+    {
+        auto variable = std::find_if(variables.cbegin(), variables.cend(),
+                                     [&var_name](ProcessVariable const& v) {
+                                         return v.getName() == var_name;
+                                     });
+
+        if (variable == variables.end())
+        {
+            OGS_FATAL(
+                "Could not find process variable '%s' in the provided "
+                "variables "
+                "list for config tag <%s>.",
+                var_name.c_str(), tag.c_str());
+        }
+        DBUG("Found process variable \'%s\' for config tag <%s>.",
+             variable->getName().c_str(), tag.c_str());
```

Comment:
This part from find_if until here is same as in the
`findProcessVariable(std::vector<ProcessVariable> const&, BaseLib::ConfigTree const&, std::string const&)`
function. Duplications should be avoided.